portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,67 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private int maxNumOfComments;
+
+  @Override
+  public void init() {
+    maxNumOfComments = 10; // default value
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      if (comments.size() < maxNumOfComments) {
+        String comment = (String) entity.getProperty("comment");
+        comments.add(comment);
+      }
+    }
+    
+    Gson gson = new Gson();
+    String commentsJson = gson.toJson(comments);
+    String json = "{\"maxNumOfComments\":" + maxNumOfComments +
+      ", \"comments\":" + commentsJson + "}";
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("comment");
+    long timestamp = System.currentTimeMillis();
+    String maxNum = request.getParameter("number-of-comments");
```

Comment:
what is this for?

Reply:
This is where we set the maximum number of comments to be displayed.